packages/language/test/fourslash/diagnostic-uris.ts

```diff
@@ -0,0 +1,26 @@
+/**
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ *
+ */
+
+/// <reference path="framework.ts" />
+
+// @filename: main.pli
+//// INTERNAL: PACKAGE EXPORTS(TB0001);
+//// TB0001: PROCEDURE() OPTIONS(MAIN);
+////   DCL  P1     POINTER;
+////   P1 = <|1:SYSNULL|>;
+//// END TB0001;
+//// END INTERNAL;
+
+testAPI.testBuilder.getDiagnostics().forEach((diagnostic) => {
+  diagnostic.uri = "file:///sysnull.pli";
+});
+testAPI.testBuilder.checkDiagnosticsURIs({ suppressTestError: true });
+verify.expectErrorCodesAt(1, code.Internal.DiagnosticURIMismatch.fullCode);
```

Comment:
Suggestion: IMO adding a synthetic diagnostic in there seems like a workaround for not using something like `verify.expectToThrow(() => testAPI.testBuilder.checkDiagnosticsURIs))`. I think we should do that instead.